docs/content/contributing/terraform/terraform-contribution-flow/_index.md

```diff
@@ -0,0 +1,313 @@
+
+1. Have/create an Azure Active Directory Service Principal with at least `Contributor` & `User Access Administrator` permissions on the Management-Group/Subscription you want to test the modules in. You might find the following links useful:
+- [Create a service principal (Azure CLI)](https://learn.microsoft.com/en-us/cli/azure/azure-cli-sp-tutorial-1) - _**Recommended**_
+- [Create a service principal (Azure Portal)](https://learn.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal)
+- [Create a service principal (PowerShell)](https://learn.microsoft.com/azure/active-directory/develop/howto-authenticate-service-principal-powershell)
+- [Find Service Principal object ID](https://learn.microsoft.com/azure/cost-management-billing/manage/assign-roles-azure-service-principals#find-your-spn-and-tenant-id)
+- [Find managed Identity Service Principal](https://learn.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-view-managed-identity-service-principal-portal)
+- Note down the following pieces of information
+- Application (Client) ID
+- Service Principal Secret (password)
+- **Optional:** Tenant ID
+- **Optional:**  Subscription ID
+
+```bash
+# Linux/MacOs
+export ARM_SUBSCRIPTION_ID=$(az account show --query id --output tsv) # or set <subscription_id>
+export ARM_TENANT_ID=$(az account show --query tenantId --output tsv) # or set <tenant_id>
+export ARM_CLIENT_ID=<client_id>
+export ARM_CLIENT_SECRET=<service_principal_password>
+
+# Windows/Powershell
+$env:ARM_SUBSCRIPTION_ID = $(az account show --query id --output tsv) # or set <subscription_id>
+$env:ARM_TENANT_ID = $(az account show --query tenantId --output tsv) # or set <tenant_id>
+$env:ARM_CLIENT_ID = "<client_id>"
+$env:ARM_CLIENT_SECRET = "<service_principal_password>"
+
+```
+
+2. Change to the root of your module repository and run `./avm docscheck` (Linux/MacOs) / `avm.bat docscheck` (Windows) to verify the container image is working as expected or needs to be pulled first. You will need this later.
+
+<img src="/Azure-Verified-Modules/img/contribution/pullImage.png" alt="Pull latest azterraform container image." width=100%>
+
+<!--
+1. Create an UAMI in your Azure test subscription.
+2. Create a role assignment for the UAMI on your test subscription, use `Contributor` role (your module might require higher privileges) such as `Owner` but we reocmmend to go with least privilege.
+
+3. Configure [federated identity credentials](https://learn.microsoft.com/en-us/entra/workload-id/workload-identity-federation-create-trust-user-assigned-managed-identity?pivots=identity-wif-mi-methods-azcli) on the user assigned managed identity. Use the GitHub `test` environment.
+
+You can use the following script to create the UAMI, role assignment and federated identity credentials:
```

Comment:
For contributors working on fork, we don't need UMIs. They don't support client secrets needed to pass creds into container as shown below.